src/tree/newtree/mod.rs

```diff
@@ -0,0 +1,57 @@
+use hpke::HPKEPrivateKey;
+
+use crate::key_packages::KeyPackageBundle;
+
+use self::binary_tree::BinaryTree;
+use self::node::Node;
+use self::provisional_tree::ProvisionalTree;
+use self::ratchet_tree::RatchetTree;
+
+use super::index::NodeIndex;
+use super::path_keys::PathKeys;
+
+mod binary_tree;
+mod node;
+mod provisional_tree;
+mod ratchet_tree;
+mod tree_secrets;
+
+#[allow(unused_variables, dead_code)]
+pub(crate) struct RatchetTreeBundle {
+    /// The index of the node that represents owner of this `RatchetTreeBundle`.
+    node_index: NodeIndex,
+
+    /// The public part of the ratchet tree.
+    public_tree: RatchetTree,
+    /// This is the HPKE private key corresponding to the `HPKEPublicKey` in the
+    /// node with index `node_index`.
+    private_key: HPKEPrivateKey,
+
+    /// A HashMap from NodeIndex to HPKEPrivateKeys in the (potentially partial)
+    /// path from this leaf.
+    path_keys: PathKeys,
```

Comment:
Naming of these members isn't great. It's close to what we have right now, which isn't great.
`path_keys`/`PathKeys` is what this is called right now. I don't actually think that this name makes a lot of sense. What path? What type of keys?
`node_index` doesn't make clear that it's about the owner of the tree. Same for `private_key`.